lib/playground.dart

```diff
@@ -162,246 +180,232 @@ class Playground implements GistContainer, GistController {
   }
 
-  void _showAboutBox() {
-    dartServices
-        .version()
-        .timeout(Duration(seconds: 2))
-        .then((VersionResponse ver) {
-      print('Dart SDK version ${ver.sdkVersion} (${ver.sdkVersionFull})');
-      print('CodeMirror: ${CodeMirrorModule.version}');
-      AboutDialog(ver.sdkVersionFull)..show();
-    }).catchError((e) {
-      AboutDialog()..show();
-    });
+  void _initButtons() {
+    newButton = MDCButton(querySelector('#new-button'))
+      ..onClick.listen((_) => _showCreateGistDialog());
+    resetButton = MDCButton(querySelector('#reset-button'))
+      ..onClick.listen((_) => _showResetDialog());
+    formatButton = MDCButton(querySelector('#format-button'))
+      ..onClick.listen((_) => _format());
+    samplesButton = MDCButton(querySelector('#samples-dropdown-button'))
+      ..onClick.listen((e) {
+        samplesMenu.open = !samplesMenu.open;
+      });
+
+    runButton = MDCButton(querySelector('#run-button'))
+      ..onClick.listen((_) {
+        _handleRun();
+      });
+    editorConsoleTab = MDCButton(querySelector('#editor-panel-console-tab'));
+    editorDocsTab = MDCButton(querySelector('#editor-panel-docs-tab'));
+    closePanelButton =
+        MDCButton(querySelector('#editor-panel-close-button'), isIcon: true);
+    moreMenuButton = MDCButton(querySelector('#more-menu-button'), isIcon: true)
+      ..onClick.listen((_) {
+        moreMenu.open = !moreMenu.open;
+      });
+    querySelector('#keyboard-button')
+        .onClick
+        .listen((_) => _showKeyboardDialog());
   }
 
-  void _resetGists() {
-    _gistStorage.clearStoredGist();
-    editableGist.reset();
-    // Delay to give time for the model change event to propagate through
-    // to the editor component (which is where `_performAnalysis()` pulls
-    // the Dart source from).
-    Timer.run(_performAnalysis);
-    _clearOutput();
+  void _initLabels() {
+    var webOutputLabelElement = querySelector('#web-output-label');
+    if (webOutputLabelElement != null) {
+      webOutputLabel = DElement(webOutputLabelElement);
+    }
   }
 
-  Future showHome(RouteEnterEvent event) async {
-    // Don't auto-run if we're re-loading some unsaved edits; the gist might
-    // have halting issues (#384).
-    bool loadedFromSaved = false;
-    Uri url = Uri.parse(window.location.toString());
+  void _initSamplesMenu() {
+    var element = querySelector('#samples-menu');
+
+    var samples = [
+      Sample('215ba63265350c02dfbd586dfd30b8c3', 'Hello World', Layout.dart),
+      Sample('e93b969fed77325db0b848a85f1cf78e', 'Int to Double', Layout.dart),
+      Sample('b60dc2fc7ea49acecb1fd2b57bf9be57', 'Mixins', Layout.dart),
+      Sample('7d78af42d7b0aedfd92f00899f93561b', 'Fibonacci', Layout.dart),
+      Sample('a559420eed617dab7a196b5ea0b64fba', 'Sunflower', Layout.html),
+      Sample('cb9b199b1085873de191e32a1dd5ca4f', 'WebSockets', Layout.html),
+      Sample('b6409e10de32b280b8938aa75364fa7b', 'Counter', Layout.flutter),
```

Comment:
The new gists I made this afternoon:

https://gist.github.com/RedBrogdon/ecb28c29c646b7f38139b1e7f44129b7 ("Draggables & physics")
https://gist.github.com/RedBrogdon/40308e0a5f47acba46ba62f4d8be2bf4 ("Implicit animations")